cmd/query/app/http_handler.go

```diff
@@ -193,6 +195,42 @@ func (aH *APIHandler) getOperationsLegacy(w http.ResponseWriter, r *http.Request
 	aH.writeJSON(w, r, &structuredRes)
 }
 
+func (aH *APIHandler) transformOTLP(w http.ResponseWriter, r *http.Request) {
+	body, err := ioutil.ReadAll(r.Body)
+	if aH.handleError(w, err, http.StatusBadRequest) {
+		return
+	}
+
+	var uiErrors []structuredError
+	batches, err := OTLP2model(body)
+
+	if aH.handleError(w, err, http.StatusInternalServerError) {
+		return
+	}
+
+	traces, err := BatchesToTraces(batches)
+
+	if aH.handleError(w, err, http.StatusInternalServerError) {
+		return
+	}
```

Comment:
why does the handler need to call two functions? the whole conversion can be encapsulated in one helper, having fewer error branches here

Reply:
I wanted to make it easier to test, would you recommend me putting it together as a single function?

Reply:
handler doesn't need to know about multiple phases of transformation. And handler is harder to test than the same two-step logic in a helper function